### PR TITLE
Allow the use of github auth tokens.

### LINKED ISSUE #6: Two-Factor Authentication
Hi, it would be very good to have an option to set the 2FA code somewhere or be queried for it. The npm module for Github already has an option to set that code, https://github.com/mikedeboer/node-github#creating-tokens-for-your-application.

### PR DESCRIPTION
**Pull request by jcvernaleo**
This allows users with two factor auth enabled to use pull-report.

This closes #6.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -32,6 +32,7 @@ $ pull-report --help
     --html                Display report as HTML
     --gh-user <username>  GitHub user name
     --gh-pass <password>  GitHub pass
+	--gh-token <token>    Github token
```
Comment (ryan-roemer) [inline README.md]: Alignment (probably need spaces, not tabs)

File: pull-report.js [hunk 2]
```diff
@@ -171,9 +172,12 @@ if (require.main === module) {
   if (!program.org) {
     throw new Error("Must specify 1+ organization names");
   }
-  if (!(program.ghUser && program.ghPass)) {
-    throw new Error("Must specify GitHub user / pass in .gitconfig or " +
-      "on the command line");
+  // If we have a token, no need for user/password
+  if (!program.ghToken) {
+    if (!(program.ghUser && program.ghPass)) {
```
Comment (ryan-roemer) [inline pull-report.js]: Combine to a single conditional please.

File: pull-report.js [hunk 3]
```diff
@@ -171,9 +172,12 @@ if (require.main === module) {
...
+      throw new Error("Must specify GitHub user / pass in .gitconfig or " +
+		      "on the command line");
```
Comment (ryan-roemer) [inline pull-report.js]: Trailing lines get indented at two per project style (no worries -- not documented :wink: )

File: pull-report.js [hunk 4]
```diff
@@ -229,11 +233,21 @@ if (require.main === module) {
...
-  });
+  // Only user basic auth if we don't have a token.
+  if (!program.ghToken) {
+    // user/pass
+    github.authenticate({
+      type: "basic",
+      username: program.ghUser,
+      password: program.ghPass,
+    });
+  }else{
```
Comment (ryan-roemer) [inline pull-report.js]: NIT: `} else {`

**Comment by ryan-roemer** [comment]
@jcvernaleo -- Thanks for the PR! Have some nits and refactors...

**Comment by jcvernaleo** [comment]
Sorry about those.  I've been spoiled by using golang (where there is a tool to enforce all those style issues that everyone uses so I don't always remember to pay attention to them).  I think that addresses all the points.

File: README.md [hunk 5]
```diff
@@ -32,6 +32,7 @@ $ pull-report --help
     --html                Display report as HTML
     --gh-user <username>  GitHub user name
     --gh-pass <password>  GitHub pass
+    -gh-token <token>    Github token
```
Comment (ryan-roemer) [inline README.md]: NIT: `--gh-token` (only has one dash right now).

File: pull-report.js [hunk 6]
```diff
@@ -171,9 +172,10 @@ if (require.main === module) {
   if (!program.org) {
     throw new Error("Must specify 1+ organization names");
   }
-  if (!(program.ghUser && program.ghPass)) {
-    throw new Error("Must specify GitHub user / pass in .gitconfig or " +
-      "on the command line");
+  // If we have a token, no need for user/password
+  if (!program.ghToken && !(program.ghUser && program.ghPass)) {
+      throw new Error("Must specify GitHub user / pass in .gitconfig or " +
+	"on the command line");
```
Comment (ryan-roemer) [inline pull-report.js]: NIT: Just one more space on the indent here.

**Comment by ryan-roemer** [comment]
@jcvernaleo -- Two nits and after that I'll try and do a release tomorrow. Thanks!

**Comment by jcvernaleo** [comment]
Sorry about that.  This time I checked it even more carefully against master so it should match up.

**Comment by ryan-roemer** [comment]
Released in `pull-report@0.1.0`

**Pull request merged**